File: leetcode/MatchPlayersAndTrainers.java
class Solution {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        
        Arrays.sort(players);
        Arrays.sort(trainers);
        
        int numberOfMatchings = 0;
        
        int i = 0, j = 0;
        
        while (i < players.length && j < trainers.length){
            if(players[i] <= trainers[j]){
                numberOfMatchings++;
                i++;
                j++;
            } else{
                j++;
            }           
        }
        
        return numberOfMatchings;
    }
}

File: leetcode/matchPlayersAndTrainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        
        players.sort()
        trainers.sort()
        
        i, j = 0, 0
        
        number_of_matching = 0
        
        while i < len(players) and j < len(trainers):
            if players[i] <= trainers[j]:
                number_of_matching += 1
                i += 1
        
            j += 1

        return number_of_matching
